### PR TITLE
add snapshot testing docs

### PR DESCRIPTION
**Pull request by hikchoi**
This PR:
- adds guide to snapshot testing

### CODE REVIEW

**Review by SeriousBug — APPROVED** [review]
Looks good to me. Would you also mind adding an item to PR review SOP that if there are changes to snapshots, the changes should be checked to make sure they are justified/make sense? I think it's easy to gloss over the snapshot changes when reviewing a PR.

File: vault/dev.process.qa.test.snapshot-testing.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+desc: ''
+updated: 1645212882651
+created: 1645212882651
+---
+
+Snapshot testing lets us assert that certain outputs of a piece of code is staying the same after changes. This is a very powerful tool because it lets us easily test for outputs that we know that will (should) be constant.
+
+That being said, snapshot tests should not be considered part of unit testing, or testing for a certain functional aspect of code. It also shouldn't replace end-to-end testing, since snapshot tests are only verifying the output, and not the behavior. The output can look the same at the end but could have gotten there in a way that you did not intend to. 
+
+While snapshot tests provides us with some great power, it has drawbacks when we don't be mindful about using them.
```
Comment (jonathanyeung) [inline vault/dev.process.qa.test.snapshot-testing.md]: nit: when we aren't mindful

**Review by kevinslin — CHANGES_REQUESTED** [review]
nice writeup. can you create a section in [[Testing|dendron://dendron.docs/dev.process.qa.test]] about snapshot testing and link to this note? I find that notes aren't discoverable unless we explicitly create links to them

**Comment by hikchoi** [comment]
suggestions applied

**Comment by hikchoi** [comment]
closing this out and updating the docs separately.

**Pull request closed without merging**